lighthouse-core/test/chromium-web-tests/test-page-devtools.sh

```diff
@@ -30,30 +32,58 @@ bash "$SCRIPT_DIR/download-content-shell.sh"
 
 bash "$SCRIPT_DIR/roll-devtools.sh"
 
-# Add test to run lighthouse in DevTools and print LHR.
-echo "
-(async function() {
-  await TestRunner.navigatePromise('$1');
+WEB_TEST_DIR="$DEVTOOLS_PATH/test/webtests/http/tests/devtools/lighthouse-run"
+
+# Remove any tests leftover from previous run.
+if [ -d "$WEB_TEST_DIR" ]; then
+  rm -rf "$WEB_TEST_DIR"
+fi
+mkdir -p "$WEB_TEST_DIR"
+
+echo "Creating test files for URLs"
+COUNTER=0
+while read url; do
+  [[ -z $url || ${url:0:1} == "#" ]] && continue
+  echo "
+  (async function() {
+    await TestRunner.navigatePromise('$url');
 
-  await TestRunner.loadModule('lighthouse_test_runner');
-  await TestRunner.showPanel('lighthouse');
+    await TestRunner.loadModule('lighthouse_test_runner');
+    await TestRunner.showPanel('lighthouse');
 
-  LighthouseTestRunner.getRunButton().click();
-  const {lhr} = await LighthouseTestRunner.waitForResults();
-  TestRunner.addResult(JSON.stringify(lhr));
+    LighthouseTestRunner.getRunButton().click();
+    const {lhr} = await LighthouseTestRunner.waitForResults();
+    TestRunner.addResult(JSON.stringify(lhr));
 
-  TestRunner.completeTest();
-})();
-" > "$DEVTOOLS_PATH/test/webtests/http/tests/devtools/lighthouse/lighthouse-run-dt.js"
+    TestRunner.completeTest();
+  })();
+  " > "$WEB_TEST_DIR/lighthouse-run-$COUNTER.js"
+  COUNTER=$[$COUNTER +1]
+done <"${1:-/dev/stdin}"
 
 set +e
-bash "$SCRIPT_DIR/web-test-server.sh" --no-show-results --time-out-ms=60000 http/tests/devtools/lighthouse/lighthouse-run-dt.js
+bash "$SCRIPT_DIR/web-test-server.sh" \
+  --no-show-results \
+  --no-retry-failures \
+  --time-out-ms=60000 \
+  --additional-driver-flag=--disable-blink-features=TrustTokens,TrustTokensAlwaysAllowIssuance \
+  http/tests/devtools/lighthouse-run
 set -e
 
+OUTPUT_DIR="$LH_ROOT/latest-run/devtools-lhrs" 
+if [ -d "$OUTPUT_DIR" ]; then
+  rm -rf "$OUTPUT_DIR"
+fi
+mkdir -p "$OUTPUT_DIR"
+
 # Copy results to latest-run folder.
 # Sometimes there will be extra output before the line with LHR. To get around this, only copy the last line with content.
-grep "lighthouseVersion" -m 1 \
-"$LH_ROOT/.tmp/layout-test-results/http/tests/devtools/lighthouse/lighthouse-run-dt-actual.txt" \
-> "$LH_ROOT/latest-run/devtools-lhr.json" 
+COUNTER=0
+for file in "$LH_ROOT/.tmp/layout-test-results/http/tests/devtools/lighthouse-run"/lighthouse-run-*-actual.txt; do
+  [ -f "$file" ] || break
+  grep "lighthouseVersion" -m 1 "$file" \
```

Comment:
I haven't seen output direction formatted like this before–any reason to not have it on a single line?